src/document/json/rga_tree_list.ts

```diff
@@ -174,7 +174,9 @@ export class RGATreeList {
 
     while (
       node!.getNext() &&
-      node!.getNext()!.getCreatedAt().after(executedAt)
+      (node!.getNext()!.getCreatedAt().after(executedAt) ||
+        (node!.getNext()!.getValue().getMovedAt() &&
+          node!.getNext()!.getValue().getMovedAt()!.after(executedAt)))
```

Comment:
I think of the logic as below:

1. When moving or inserting an element in the same position, it must be aligned by the time the element was positioned.
2. The time the element is positioned can be obtained using the time the element is inserted(`createdAt`) and the time it moves(`movedAt`).
3. Since the element can be moved after the element is inserted, the movement time must be considered first.

Based on these, I coded it like this:

```suggestion
      node!.getNext()!.getPositionedAt().after(executedAt)
```

for `RGATreeListNode.getPosiontedAt()`:

```typescript
  /**
   * `getPositionedAt` returns time this element was positioned in the array.
   */
  public getPositionedAt(): TimeTicket {
    const movedAt = this.value.getMovedAt();
    if (movedAt) {
      return movedAt;
    }

    return this.value.getCreatedAt();
  }
```